pkg/result/ignore.go

```diff
@@ -0,0 +1,174 @@
+package result
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/samber/lo"
+	"golang.org/x/exp/slices"
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v3"
+
+	"github.com/aquasecurity/trivy/pkg/clock"
+	"github.com/aquasecurity/trivy/pkg/log"
+)
+
+// IgnoreFinding represents an item to be ignored.
+type IgnoreFinding struct {
+	// ID is the identifier of the vulnerability, misconfiguration, secret, or license.
+	// e.g. CVE-2019-8331, AVD-AWS-0175, etc.
+	// required: true
+	ID string `yaml:"id"`
+
+	// Paths is the list of file paths to be ignored.
+	// If Paths is not set, the ignore finding is applied to all files.
+	// required: false
+	Paths []string `yaml:"paths"`
+
+	// ExpiredAt is the expiration date of the ignore finding.
+	// If ExpiredAt is not set, the ignore finding is always valid.
+	// required: false
+	ExpiredAt time.Time `yaml:"expired_at"`
+
+	// Statement describes the reason for ignoring the finding.
+	// required: false
+	Statement string `yaml:"statement"`
+}
+
+type IgnoreFindings []IgnoreFinding
+
+func (findings IgnoreFindings) Match(path, id string) bool {
+	for _, finding := range findings {
+		if len(finding.Paths) != 0 && !slices.Contains(finding.Paths, path) {
```

Comment:
Perhaps we want to use `wildcard` for paths?

Reply:
don't forget: we need to update docs if we don't add wildcards.

Reply:
Added https://github.com/aquasecurity/trivy/pull/5070/commits/c624be43c9c7339e37ac7c2a7974599d4cbbfe78